src/envoy/http/authn/origin_authenticator.cc

```diff
@@ -42,45 +45,30 @@ void OriginAuthenticator::run() {
                   "Principal is binded to origin, but no method specified in "
                   "policy {}",
                   policy_.DebugString());
-        onMethodDone(nullptr, false);
         break;
       case iaapi::PrincipalBinding::USE_PEER:
         // On the other hand, it's ok to have no (origin) methods if
         // rule USE_SOURCE
-        onMethodDone(nullptr, true);
+        success = true;
         break;
       default:
         // Should never come here.
         ENVOY_LOG(error, "Invalid binding value for policy {}",
                   policy_.DebugString());
         break;
     }
-    return;
   }
-  runMethod(policy_.origins(0), [this](const Payload* payload, bool success) {
-    onMethodDone(payload, success);
-  });
-}
 
-void OriginAuthenticator::runMethod(
```

Comment:
1. here runMethod only calls validateJwt, why do we still need to keep it ?
2. sg, looks like we can remove DoneCallback from AuthenticatorBase in http_filter.cc; I may prefer to do it in a separated PR to prevent this one getting larger, what do you think ?

Reply:
Sorry, github diff gives me the impression that there are more in that function. In that case, it's ok to remove the function.

I think it's better to remove all callbacks at once, but it's up to you.

Reply:
done, looks much cleaner now :)